### PR TITLE
Reintroduce governanceWithdraw function

### PR DESCRIPTION
**Pull request by cwsnt**

### CODE REVIEW

**Review by ororopickpocket — CHANGES_REQUESTED** [review]

File: contracts/governance/Vesting/fouryear/FourYearVestingLogic.sol [hunk 1]
```diff
@@ -328,7 +328,11 @@ contract FourYearVestingLogic is IFourYearVesting, FourYearVestingStorage, Appro
 
             /// @dev Withdraw if > 0
             if (stake > 0) {
-                staking.withdraw(stake, i, receiver);
+                if (isGovernance) {
+                    staking.governanceWithdraw(stake, i, receiver);
+                } else {
+                    staking.withdraw(stake, i, receiver);
+                }
```
[the remark this answers is not in the document]
Comment (cwsnt) [reply contracts/governance/Vesting/fouryear/FourYearVestingLogic.sol]: updated in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/487/commits/a67d64a582875af2df507533eab1953de0e24818)

File: tests/staking/PauseStaking.test.js [hunk 2]
```diff
@@ -480,6 +480,188 @@ contract("Staking", (accounts) => {
...
+                previousAmount.sub(new BN(toStake).mul(new BN(2))).toString(),
+                amountAfterStake.toString()
+            );
+            assert.equal(previousAmount.toString(), amount.toString());
+
+            let vestingBalance = await staking.balanceOf(vesting.address);
+            expect(vestingBalance).to.be.bignumber.equal(new BN(0));
+        });
+
+        it("governanceWithdrawVesting and cancelTeamVesting function should not be overlapped", async () => {
```
[the remark this answers is not in the document]
Comment (cwsnt) [reply tests/staking/PauseStaking.test.js]: updated in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/487/commits/a67d64a582875af2df507533eab1953de0e24818)

**Review by ororopickpocket — APPROVED** [review]

File: contracts/governance/Vesting/VestingLogic.sol [hunk 3]
```diff
@@ -116,7 +116,9 @@ contract VestingLogic is IVesting, VestingStorage, ApprovalReceiver {
      * @dev Can be called only by owner.
      * */
     function governanceWithdrawTokens(address receiver) public {
-        revert("deprecated, use cancelTeamVesting from the staking contract");
+        require(msg.sender == address(staking), "unauthorized");
```
Comment (tjcloa) [inline contracts/governance/Vesting/VestingLogic.sol:119]: missing a warning comment that this function should not be used with sovryn protocol

File: contracts/governance/Staking/interfaces/IStaking.sol [hunk 4]
```diff
@@ -580,6 +580,27 @@ interface IStaking {
         address receiver
     ) external;
 
+    /**
```
Comment (tjcloa) [inline contracts/governance/Staking/interfaces/IStaking.sol:583]: missing a warning comment not to use in sovryn

File: contracts/mockup/modules/IStakingModuleBlockMockup.sol [hunk 5]
```diff
@@ -515,6 +515,19 @@ interface IStakingModuleBlockMockup {
         address receiver
     ) external;
 
+    /**
```
Comment (tjcloa) [inline contracts/mockup/modules/IStakingModuleBlockMockup.sol:518]: despite this is a mocking contract it doesn't hurt to add a deprecation warning for sovryn

**Review by tjcloa — APPROVED** [review]
pls add missing deprecation warnings not to use governance vesting withdrawals in the code to avoid confusion in the future - see inline comments

**Pull request merged**